test/SuperUserSpark/Compiler/Gen.hs

```diff
@@ -55,6 +51,6 @@ instance GenUnchecked CompileError where
             [ PreCompileErrors <$> genUnchecked
             , DuringCompilationError <$> genUnchecked
             ]
+    shrinkUnchecked _ = []
```

Comment:
This seems very wrong. Why did you add this?

Reply:
Text.Parsec.Error.ParseError has no GenUnchecked and GenValid instances. Therefore, you can't derive these instances for ParseError (which you indeed didn't do, not sure whether you want to do this). Since ParseError is used in CompileError and CompileState, they can't derive shrinkUnchecked from Generic. This is why I added this line.
Note: The version of genvalidity which was used before I upgraded the bounds, didn't include any shrinking function yet. This is why I only needed to add this line now.
Possible solutions:
1. define this default shrinkUnchecked
2. define a GenValid instance for Text.Parsec.Error.ParseError, for your wrapper around that type, and derive shrinkUnchecked for CompileError and CompileState.
Which one do you prefer?

Reply:
Just leave this then, it's fine.
I forgot that shrinkUnchecked is new.